pkg/cmd/roachtest/tests/backup_fixtures.go

```diff
@@ -8,217 +8,219 @@ package tests
-	if !bd.sp.scheduledBackupSpecs.ignoreExistingBackups {
-		// This check allows the roachtest to fail fast, instead of when the
-		// scheduled backup cmd is issued.
-		require.False(bd.t, bd.checkForExistingBackupCollection(ctx), fmt.Sprintf("existing backup in collection %s", bd.sp.scheduledBackupSpecs.backupCollection()))
-	}
-}
-
-// checkForExistingBackupCollection returns true if there exists a backup in the collection path.
-func (bd *backupDriver) checkForExistingBackupCollection(ctx context.Context) bool {
-	collectionQuery := fmt.Sprintf(`SELECT count(*) FROM [SHOW BACKUPS IN %s]`,
-		bd.sp.scheduledBackupSpecs.backupCollection())
 	conn := bd.c.Conn(ctx, bd.t.L(), 1)
-	sql := sqlutils.MakeSQLRunner(conn)
-	var collectionCount int
-	sql.QueryRow(bd.t, collectionQuery).Scan(&collectionCount)
-	return collectionCount > 0
-}
 
-func (bd *backupDriver) assertCorrectCockroachBinary(ctx context.Context) {
-	binaryQuery := "SELECT value FROM crdb_internal.node_build_info WHERE field = 'Version'"
-	conn := bd.c.Conn(ctx, bd.t.L(), 1)
-	sql := sqlutils.MakeSQLRunner(conn)
-	var binaryVersion string
-	sql.QueryRow(bd.t, binaryQuery).Scan(&binaryVersion)
-	if bd.sp.scheduledBackupSpecs.version != fixtureFromMasterVersion {
-		require.Equal(bd.t, bd.sp.scheduledBackupSpecs.version, binaryVersion, "cluster not running on expected binary")
-	} else {
-		require.Contains(bd.t, binaryVersion, "dev")
-	}
+	// Work around an issue with import where large imports can bottlneck on
+	// snapshots.
+	_, err = conn.Exec("set cluster setting kv.snapshot_rebalance.max_rate = '256 MiB'")
+	require.NoError(bd.t, err)
 }
 
 func (bd *backupDriver) initWorkload(ctx context.Context) {
-	if bd.sp.initWorkloadViaRestore == nil {
-		bd.t.L().Printf(`Initializing workload`)
-		bd.sp.scheduledBackupSpecs.workload.init(ctx, bd.t, bd.c, bd.sp.hardware)
-		return
-	}
-	computedRestoreSpecs := restoreSpecs{
-		hardware:               bd.sp.hardware,
-		backup:                 makeBackupSpecs(bd.sp.initWorkloadViaRestore.backup, bd.sp.scheduledBackupSpecs.backupSpecs),
-		restoreUptoIncremental: bd.sp.initWorkloadViaRestore.restoreUptoIncremental,
-	}
-	restoreDriver := makeRestoreDriver(bd.t, bd.c, computedRestoreSpecs)
-	bd.t.L().Printf(`Initializing workload via restore`)
-	restoreDriver.getAOST(ctx)
-	// Only restore the database because a cluster restore will also restore the
-	// scheduled_jobs system table, which will automatically begin any backed up
-	// backup schedules, which complicates fixture generation.
-	target := fmt.Sprintf("DATABASE %s", restoreDriver.sp.backup.workload.DatabaseName())
-	require.NoError(bd.t, restoreDriver.run(ctx, target))
+	bd.t.L().Printf("importing tpcc with %d warehouses", bd.sp.fixture.ImportWarehouses)
+
+	urls, err := bd.c.InternalPGUrl(ctx, bd.t.L(), bd.c.Node(1), roachprod.PGURLOptions{})
+	require.NoError(bd.t, err)
+
+	cmd := roachtestutil.NewCommand("./cockroach workload fixtures import tpcc").
+		Arg("%q", urls[0]).
```

Comment:
could tpcc init be faster if it connects across several nodes? like the c2c tests?

Reply:
Nope. The `workload fixtures import` command works by creating an import job for each TPCC table. So the workload -> db connections don't carry any of the import data.